sdks/wasm/src/library_mono.js

```diff
@@ -41,8 +41,15 @@ var MonoSupportLib = {
 				var decoder = typeof TextDecoder !== 'undefined' ? new TextDecoder ('utf-16le') : undefined;
 
 				var str = "";
-				if (decoder)
-					str = decoder.decode(Module.HEAPU8.subarray(start, end));
+            			if (decoder) {
+					// When threading is enabled, TextDecoder does not accept a view of a 
+					// SharedArrayBuffer, we must make a copy of the array first.
+                			var subArray = Module.wasmMemory.buffer instanceof SharedArrayBuffer
```

Comment:
```suggestion
                			var subArray = Module.buffer instanceof SharedArrayBuffer
```
It looks like this should be testing Module.buffer for the general case?  Also stick with tabs for indentation and `} else {` instead of the newline after the closing brace

Reply:
I'd personally test Module.HEAPU8 since that's what you use immediately after

Reply:
There's no `Module.buffer` that I could find, and Module.HEAPU8 is of type `Uint8Array` for which I did not find a way to test if it's based on a `SharedArrayBuffer` or not.

Reply:
TypedArray instances have a ```.buffer``` property you can use:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/buffer

Reply:
Thanks! I did not look at the right place in the debugger. Will update.